common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java

```diff
@@ -177,27 +180,31 @@ public abstract AcquireTokenResult acquireDeviceCodeFlowToken(@SuppressWarnings(
         }
 
         builder.setClientId(parameters.getClientId())
-                .setRedirectUri(parameters.getRedirectUri())
-                .setCorrelationId(correlationId);
+                .setRedirectUri(parameters.getRedirectUri());
+
+        if (builder instanceof MicrosoftAuthorizationRequest.Builder) {
+            ((MicrosoftAuthorizationRequest.Builder) builder).setCorrelationId(correlationId);
```

Comment:
Not for you necessarily... I'm sure I created this originally, but i think we should just move this accessor to the base class.